lib/utils.js

```diff
@@ -58,7 +58,7 @@ exports.forEach = function(arr, fn, scope){
 exports.map = function(arr, fn, scope){
   var result = [];
   for (var i = 0, l = arr.length; i < l; i++)
-    result.push(fn.call(scope, arr[i], i));
+    result.push(fn.call(scope, arr[i], i, arr));
```

Comment:
looks like this is just being done so the test has access to the original array? if so, just make a var for the array in the test and get rid of this